beam/task/types/src/Rhyolite/Task/Beam.hs

```diff
@@ -0,0 +1,39 @@
+{-# Language DeriveAnyClass #-}
+{-# Language DeriveGeneric #-}
+{-# Language FlexibleInstances #-}
+{-# Language FunctionalDependencies #-}
+{-# Language KindSignatures #-}
+{-# Language RankNTypes #-}
+{-# Language StandaloneDeriving #-}
+{-# Language TemplateHaskell #-}
+{-|
+Description : Contains definitions for tasks, and tables that have tasks.
+
+For more details on how a Task is defined, refer to the Task data type below.
+
+This module should be used when there is a list of tasks to be processed, and a limited number of workers available
+to process those tasks. The workers will pick off tasks one by one, and store the results back in the table of tasks.
+-}
+module Rhyolite.Task.Beam where
+
+import Control.Lens
+
+import Database.Beam
+
+-- | The 'Task' type describes how to use a database table to check out tasks
+-- and report the results back.
+data Task be table payload checkout result = Task
+  { _task_filter :: forall s. table (QExpr be s) -> QExpr be s Bool
+  -- ^ How to filter unclaimed task rows
```

Comment:
Could the wording here be improved to clarify what "unclaimed" means? AIUI it's something like:

```haskell
  { _task_filter :: forall s. table (QExpr be s) -> QExpr be s Bool
  -- ^ Filter for those rows which haven't been picked up by a worker
```

do I have that right?